rotkehlchen/chain/ethereum/modules/eigenlayer/decoder.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any
+
+from rotkehlchen.chain.ethereum.modules.eigenlayer.constants import (
+    CPT_EIGENLAYER,
+    DEPOSIT_TOPIC,
+    EIGENLAYER_CPT_DETAILS,
+    EIGENLAYER_STRATEGY_MANAGER,
+)
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.resolver import ethaddress_to_identifier
+from rotkehlchen.history.events.structures.evm_event import EvmProduct
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address
+
+
+class EigenlayerDecoder(DecoderInterface):
+
+    def _decode_deposit(self, context: DecoderContext) -> DecodingOutput:
```

Comment:
hmm this feels incomplete. Please at least handle withdrawals too and whatever else may be there